app/models/aws_provider.rb

```diff
@@ -68,7 +68,11 @@ def wait_for_instances(state, ids)
  
   def create_instance 
     connect 
-    instance_name = CloudUtils.generate_instance_name('cm-')
+    if @instance_name.nil?
+       name = CloudUtils.generate_instance_name('cm-')
```

Comment:
suggesting using instance_name directly instead of name as a transitive variable. Using the extra name variable doesn't seem to offer any  benefits here.